### PR TITLE
feat: Addition of 'Practice Problems' section to sidebar with topic-specific question pages

### PR DESCRIPTION
**Pull request by Chaitanya-45**
Fixes: #46 
- Added a new 'Practice Problems' section in the sidebar
- Created individual pages for questions covering various C++ topics:
  - Introductory Problems
  - Control Structure Problems
  - Arrays
  - Strings
  - Functions
  - Structures and Pointers
  - Sorting
  - Standard Template Library (STL)

### CODE REVIEW

**Comment by subhadipbhowmik** [comment]
> Fixes: #46
> 
>     * Added a new 'Practice Problems' section in the sidebar
> 
>     * Created individual pages for questions covering various C++ topics:
>       
>       * Introductory Problems
>       * Control Structure Problems
>       * Arrays
>       * Strings
>       * Functions
>       * Structures and Pointers
>       * Sorting
>       * Standard Template Library (STL)

Could You Please Implement in the Blog section not in Docs?

**Comment by Chaitanya-45** [comment]
@subhadipbhowmik Please check the update, I've moved the practice problems to the blogs section and added respective tags categorizing them based on the concept and coding platform.

File: blog/2024-06-03-STL-problems/STL-problems.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+slug: STL-problems
```
Comment (subhadipbhowmik) [inline blog/2024-06-03-STL-problems/STL-problems.md:2]: in slug make STL to lowercase

File: blog/2024-06-03-structure-probelms/structure-problems.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+slug: structure-problems
+title: Structures and Pointers Problems
+authors: [chaitanya]
+tags: ["30daysofcpp", "struct", "pointer","practice problems","hackerrank","geeksforgeeks","codechef"]
```
Comment (subhadipbhowmik) [inline blog/2024-06-03-structure-probelms/structure-problems.md:5]: no space in tags use -

**Comment by Chaitanya-45** [comment]
@subhadipbhowmik I've made the suggested changes, please check.

**Comment by subhadipbhowmik** [comment]
> @subhadipbhowmik I've made the suggested changes, please check.



> @subhadipbhowmik I've made the suggested changes, please check.

Just wait a while

**Pull request merged**